src/pages/home/report/ReportActionItem.js

```diff
@@ -70,59 +67,45 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = React.useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const previousDraftMessage = React.useRef('');
+    const textInput = React.useRef();
+    const popoverAnchor = React.useRef();
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return this.props.displayAsGroup !== nextProps.displayAsGroup
-            || this.props.draftMessage !== nextProps.draftMessage
-            || this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction
-            || this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU
-            || this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker
-            || !_.isEqual(this.props.action, nextProps.action)
-            || this.state.isContextMenuActive !== nextState.isContextMenuActive;
-    }
+    React.useEffect(() => {
+        previousDraftMessage.current = props.draftMessage;
+    }, [props.draftMessage]);
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.draftMessage || !this.props.draftMessage) {
+    React.useEffect(() => {
+        if (previousDraftMessage.current || !props.draftMessage) {
             return;
         }
 
         // Only focus the input when user edits a message, skip it for existing drafts being edited of the report.
         // There is an animation when the comment is hidden and the edit form is shown, and there can be bugs on different mobile platforms
         // if the input is given focus in the middle of that animation which can prevent the keyboard from opening.
-        focusTextInputAfterAnimation(this.textInput, 100);
-    }
+        focusTextInputAfterAnimation(textInput.current, 100);
+    }, [props.draftMessage]);
 
-    checkIfContextMenuActive() {
-        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
-    }
+    const checkIfContextMenuActive = () => {
+        setIsContextMenuActive(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    };
 
     /**
      * Show the ReportActionContextMenu modal popover.
      *
      * @param {Object} [event] - A press event.
      */
-    showPopover(event) {
+    const showPopover = (event) => {
```

Comment:
I think we should standardize on `function`

Reply:
<img width="488" alt="Screenshot 2023-03-31 at 4 26 35 PM" src="https://user-images.githubusercontent.com/31285285/229177066-11d555a7-752c-49f7-bb26-ef0097f0f2a5.png">

It seems we have an existing linter rule that doesn't let us pass functions that use `function` as props. So let's stick with the arrow syntax?

Reply:
ah, actually, no. There was a conversation about this here: https://expensify.slack.com/archives/C01GTK53T8Q/p1680096510744619

The solution is to either:

1. Move the function out of the component if it has no dependencies
2. Wrap it in `useCallback()` if it does have dependencies